ui/src/app/shared/namespaces.ts

```diff
@@ -0,0 +1,64 @@
+const managedNamespaceKey = 'managedNamespace';
+const userNamespaceKey = 'userNamespace';
+const currentNamespaceKey = 'current_namespace';
+
+// just a temp function, this gets set in app-router
+let onNamespaceChange = (x: string) => x;
+export function setOnNamespaceChange(f: any) {
+    onNamespaceChange = f;
+}
+
+function fixLocalStorageString(x: string): string {
+    // empty string is valid, so we cannot use `truthy`
+    if (x == null || x == 'null' || x == 'undefined') {
+        return undefined; // explicitly return undefined
+    }
+    return x;
+}
+
+export function setUserNamespace(value: string) {
+    if (value) {
+        localStorage.setItem(userNamespaceKey, value);
+    } else {
+        localStorage.removeItem(userNamespaceKey);
+    }
+}
+
+function getUserNamespace() {
+    return fixLocalStorageString(localStorage.getItem(userNamespaceKey));
+}
+
+export function setManagedNamespace(value: string) {
+    if (value) {
+        localStorage.setItem(managedNamespaceKey, value);
+    } else {
+        localStorage.removeItem(managedNamespaceKey);
+    }
+}
+
+export function getManagedNamespace() {
+    return fixLocalStorageString(localStorage.getItem(managedNamespaceKey));
+}
+
+export function setCurrentNamespace(value: string) {
+    if (value != null) {
+        localStorage.setItem(currentNamespaceKey, value);
+    } else {
+        localStorage.removeItem(currentNamespaceKey);
+    }
+    onNamespaceChange(getCurrentNamespace());
+}
+
+export function getCurrentNamespace() {
+    return fixLocalStorageString(localStorage.getItem(currentNamespaceKey)) ?? (getUserNamespace() || getManagedNamespace());
```

Comment:
ack part of this is accidentally taken from #13347 (merge conflict/stash issues); I'm gonna mark this as draft temporarily as a result. Will rebase once that's merged

Reply:
Rebased